### PR TITLE
Few fixes

### PR DESCRIPTION
**Pull request by jacquesbh**
- feat: Use Context objects instead of deprecated parameters
- feat: Better way to get the default value of the repo's name

### CODE REVIEW

File: castor.php [hunk 1]
```diff
@@ -42,52 +43,56 @@ function setup(
...
+    run('symfony composer config --unset authors', context: $composerContext);
+    run('symfony composer config --unset keywords', context: $composerContext);
+    run('symfony composer config extra.symfony.allow-contrib true', context: $composerContext);
+    run('symfony composer require --no-scripts php="^' . $phpVersion . '"', context: $composerContext);
 
     # Add scripts in composer.json
-    run('symfony composer config scripts.phpcs "php-cs-fixer fix --allow-risky=yes"', path: 'apps/sylius/');
-    run('symfony composer config scripts.phpmd "phpmd src/,plugins/ ansi phpmd.xml --exclude src/Migrations"', path: 'apps/sylius/');
+    run('symfony composer config scripts.phpcs "php-cs-fixer fix --allow-risky=yes"', workingDirectory: 'apps/sylius/');
+    run('symfony composer config scripts.phpmd "phpmd src/,plugins/ ansi phpmd.xml --exclude src/Migrations"', workingDirectory: 'apps/sylius/');
```
Comment (delyriand) [inline castor.php:72]: **❔ question:** why not use `context: $composerContext` for these two commands as for the others?‏

File: castor.php [hunk 2]
```diff
@@ -42,52 +51,56 @@ function setup(
...
 
     # Fix for sylius and doctrine conflict
     io()->info('Add conflict for doctrine/orm in order to fix an issue in Sylius.');
-    run("cat composer.json | jq --indent 4 '.conflict += {\"doctrine/orm\": \">= 2.15.2\"}' > composer.json.tmp", path: 'apps/sylius/');
-    run('mv composer.json.tmp composer.json', path: 'apps/sylius/');
+    run("cat composer.json | jq --indent 4 '.conflict += {\"doctrine/orm\": \">= 2.15.2\"}' > composer.json.tmp", context: $composerContext);
+    run('mv composer.json.tmp composer.json', context: $composerContext);
     io()->info('Run composer update after updating the composer.json file.');
-    run('symfony composer update', timeout: false, path: 'apps/sylius/');
+    run('symfony composer update', context: $composerContext);
```
Comment (delyriand) [inline castor.php:103]: **❔ question:** you go from a false timeout to DEFAULT_TIMEOUT_COMPOSER_PROCESS with this context. won't this present problems during the run?‏

File: castor.php [hunk 3]
```diff
@@ -17,6 +18,14 @@
...
 const SUGGESTED_SYLIUS_VERSION = '1.12';
 
+#[AsTask]
+function test(): void
+{
+    dump(capture('gh repo view --json nameWithOwner --jq .nameWithOwner', onFailure: 'monsieurbiz/project'));
+    $repo = strtolower(capture('gh repo view --json nameWithOwner --jq .nameWithOwner', onFailure: 'monsieurbiz/project'));
+
+    dump($repo);
+}
```
Comment (delyriand) [inline castor.php:28]: **💣 chore:** remove the test task‏

**Review by delyriand — CHANGES_REQUESTED** [review]

File: castor.php (hunk 2 above)
Comment (jacquesbh) [reply castor.php:103]: Non, I don't think so. It's composer anyway, we need this step.

File: castor.php (hunk 3 above)
Comment (jacquesbh) [reply castor.php]: Arf. I was being careful not to include it, then after your first review I made a mistake and did include it… Damn me.

**Review by delyriand — APPROVED** [review]

**Pull request merged**